modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/layer/MBLayer.java

```diff
@@ -301,6 +325,47 @@ public JSONObject paint(){
     public JSONObject getPaintProperties(){
         return new JSONObject();
     }
+
+    /**
+     * Transforms a given {@link MBLayer} to a GeoTools {@link FeatureTypeStyle}.
+     *
+     * @param layer The MBLayer to transform.
+     * @param minScaleDenominator Used to determine zoom level restructions for generated rules
+     * @param maxScaleDenominator Used to determine zoom level restructions for generated rules
+     * @return A feature type style from the provided layer.
+     */
+    public FeatureTypeStyle transform(MBStyle styleContext, Double minScaleDenominator, Double maxScaleDenominator) {
+        // Would prefer to accept zoom levels here (less concepts in our API)
+        // If we accept zoom levels we may be able to reduce, and return a list of FeatureTypeStyles
+        // (with the understanding that the list may be empty if the MBLayer does not contribute any content
+        //  at a specific zoom level range)
+        FeatureTypeStyle style = transform(styleContext);
+        for (Rule rule : style.rules()) {
```

Comment:
setting min/max on rules after constructions seems wrong (code operating on different levels of detail). Consider passing min/max into the transform( styleContext, minZoomLevel, maxZoomLevel).